govuk_chat_evaluation/jailbreak_guardrails/generate.py

```diff
@@ -0,0 +1,40 @@
+import asyncio
+from pathlib import Path
+from ..dataset_generation import generate_dataset, run_rake_task
+from ..file_system import jsonl_to_models, write_generated_to_output
+from .evaluate import EvaluationResult
+from pydantic import BaseModel
+from typing import List
+
+
+class GenerateInput(BaseModel):
+    question: str
+    expected_outcome: bool
+
+
+def generate_and_write_dataset(input_path: str, provider: str, output_dir: Path):
+    models = jsonl_to_models(Path(input_path), GenerateInput)
+    generated = generate_models_to_evaluation_models(provider, models)
+    return write_generated_to_output(output_dir, generated)
+
+
+def generate_models_to_evaluation_models(
+    provider: str, models: List[GenerateInput]
+) -> List[EvaluationResult]:
+    """Asynchronously run rake tasks for each GenerateInput instance to
+    generate a result"""
+
+    async def generate_to_evaluate(input: GenerateInput):
+        env = {"INPUT": input.question}
+        result = await run_rake_task(
+            f"evaluation:generate_jailbreak_guardrail_response[{provider}]",
+            env,
+        )
```

Comment:
[question] why is the input value passed as an `env` (which I naively interpret as an enviornment variable?

Reply:
It is environment variables that the rake task is called with. I think calling it `env_vars` might make that clearer.